rs/bitcoin/kyt/src/fetch/tests.rs

```diff
@@ -0,0 +1,489 @@
+    }
+    fn expect_get_tx_with_reply(&self, reply: Result<Transaction, HttpGetTxError>) {
+        self.replies.borrow_mut().push_back(reply)
+    }
+    fn refill_cycles(&self, cycles: u128) {
+        *self.available_cycles.borrow_mut() = cycles;
+    }
+    fn cycles_accepted(&self) -> u128 {
+        *self.accepted_cycles.borrow()
+    }
+}
+
+fn mock_txid(v: u8) -> Txid {
+    Txid::from([v; 32])
+}
+
+fn mock_transaction() -> Transaction {
+    Transaction {
+        version: Version::ONE,
+        lock_time: LockTime::ZERO,
+        input: Vec::new(),
+        output: Vec::new(),
+    }
+}
+
+fn mock_transaction_with_outputs(num_outputs: usize) -> Transaction {
+    let mut tx = mock_transaction();
+    let output = (0..num_outputs)
+        .map(|i| TxOut {
+            value: Amount::ONE_SAT,
+            script_pubkey: ScriptBuf::new_p2pkh(&PubkeyHash::from_slice(&[i as u8; 20]).unwrap()),
+        })
+        .collect();
+    tx.output = output;
+    tx
+}
+
+fn mock_transaction_with_inputs(input_txids: Vec<(Txid, u32)>) -> Transaction {
+    let mut tx = mock_transaction();
+    let input = input_txids
+        .into_iter()
+        .enumerate()
+        .map(|(i, (txid, vout))| TxIn {
+            previous_output: OutPoint {
+                txid: bitcoin::Txid::from_slice(txid.as_ref()).unwrap(),
+                vout,
+            },
+            script_sig: ScriptBuf::from_bytes(vec![i as u8; 32]),
+            sequence: Sequence::ZERO,
+            witness: Witness::new(),
+        })
+        .collect();
+    tx.input = input;
+    tx
+}
+
+#[tokio::test]
+async fn test_mock_env() {
+    // Test cycle mock functions
+    let env = MockEnv::new(CHECK_TRANSACTION_CYCLES_REQUIRED);
+    assert_eq!(
+        env.cycles_accept(CHECK_TRANSACTION_CYCLES_SERVICE_FEE),
+        CHECK_TRANSACTION_CYCLES_SERVICE_FEE
+    );
+    let available = env.cycles_available();
+    assert_eq!(
+        available,
+        CHECK_TRANSACTION_CYCLES_REQUIRED - CHECK_TRANSACTION_CYCLES_SERVICE_FEE
+    );
+    assert_eq!(
+        env.cycles_accept(CHECK_TRANSACTION_CYCLES_REQUIRED),
+        available
+    );
+
+    // Test get_tx mock function
+    let env = MockEnv::new(0);
+    let txid = mock_txid(0);
+    env.expect_get_tx_with_reply(Ok(mock_transaction()));
+    let result = env.http_get_tx(txid, INITIAL_MAX_RESPONSE_BYTES).await;
+    assert!(result.is_ok());
+    env.assert_get_tx_call(txid, INITIAL_MAX_RESPONSE_BYTES);
+    env.assert_no_more_get_tx_call();
+}
+
+#[test]
+fn test_try_fetch_tx() {
+    let mut state = MockState::new();
+    let env = MockEnv::new(CHECK_TRANSACTION_CYCLES_REQUIRED);
+    let txid_0 = mock_txid(0);
+    let txid_1 = mock_txid(1);
+
+    // case Fetched
+    let fetched_0 = FetchTxStatus::Fetched(FetchedTx {
+        tx: mock_transaction(),
+        input_addresses: vec![None],
+    });
+    state.set_fetch_status(txid_0, fetched_0.clone());
+    assert!(matches!(
+        env.try_fetch_tx(&state, txid_0),
```

Comment:
What about testing an unknown transaction ID, e.g.. `env.try_fetch_tx(&state, unknown_txid)`?

Reply:
That is the last case being tested in this `test_try_fetch_tx()` function.

Reply:
sorry I missed it.